### ISSUE TITLE
Fix FP on S4123 (`no-invalid-await`): invalid js-doc with return type

### ISSUE THREAD

**Issue by vilchik-elena** [SonarSource/SonarJS#2859]
FP is reported in the following case (`.js` file)
```js
    /**
    * @returns {Object}
    */
    const foo = async () => {
      return "foo";
    };
    
    await foo(); // reported FP
```

To reproduce in UT use `RuleTesterJsWithTypes` as it's setting production-like tsconfig for JS files with `noImplicitAny` option. 

Without this option compiler will infer `any` as return type (probably as there is some conflicting info, Promise vs Object), while with this option it prefers type from js-doc (`Object`) thus we report a FP.

We discovered such FP on Peach (Kibana project), no user provided such feedback. Many users complain about FPs for this rule but so far other reasons were not identifieed.

**Comment by robertpatrick** [comment]
Might as well turn this rule off since the implementation is naive.  All async functions always return a Promise but for whatever reason, this rule doesn't seem to understand this fact.

**Comment by vilchik-elena** [comment]
@robertpatrick Please open a new issue with the code examples reproducing false positive issues you are experiencing, it will help us improve the rule

**Comment by robertpatrick** [comment]
@vilchik-elena The example shown in this issue should reproduce the problem.  If not, open the https://github.com/oracle/weblogic-toolkit-ui project in your favorite IDE and pick just about any JavaScript file (e.g., `electron/app/main.js`).

**Comment by vilchik-elena** [comment]
@robertpatrick are you using latest SonarLint? This rule should not be able anymore to report anything in JS files as typing information is not reliable.

**Comment by robertpatrick** [comment]
I am using the latest version of IntelliJ and the latest version of their Sonar Lint plugin…what am I missing?

**Comment by vilchik-elena** [comment]
@robertpatrick are you using connected mode? which version of SonarLint are you using?

**Comment by Wohops** [comment]
Migrated to Jira. New ticket number: JS-125
